File: test/comparison-cases/complex-styling/case.ts
#!/usr/bin/env bun
import { rgbColor, string } from '../../../../src/index.js';

// Mock process.stdout.write to capture output
const originalWrite = process.stdout.write;
let capturedOutput = '';
// biome-ignore lint/suspicious/noExplicitAny: Test utility function requires any for stdout mock
(process.stdout.write as any) = (chunk: string | Uint8Array) => {
  capturedOutput += chunk.toString();
  return true;
};

try {
  const styled = string('Complex')
    .foreground(rgbColor('#FF0000'))
    .background(rgbColor('#00FF00'))
    .bold()
    .underline()
    .italic();
  process.stdout.write(styled.toString());

  console.log(capturedOutput);
} finally {
  process.stdout.write = originalWrite;
}

File: test/comparison-cases/complex-styling/case.go
package main

import (
	"fmt"
	"strings"

	"github.com/muesli/termenv"
)

var output strings.Builder

func init() {
	// Redirect output to capture it
	termenv.SetDefaultOutput(termenv.NewOutput(&output, termenv.WithProfile(termenv.TrueColor)))
}

func main() {
	
        s := termenv.String("Complex").
          Foreground(termenv.RGBColor("#FF0000")).
          Background(termenv.RGBColor("#00FF00")).
          Bold().
          Underline().
          Italic()
        output.WriteString(s.String())
        
	fmt.Print(output.String())
}
